ext/spl/tests/SplFixedArray_serialize.phpt

```diff
@@ -24,10 +26,23 @@ var_dump($unser[0], $unser[1], $unser[2], $unser[3], $unser[4]);
 
 $unser[4] = 'quux';
 var_dump($unser[4]);
+var_dump($unser->foo);
+
+$array = new SplFixedArray(5);
+$array->__unserialize([
+    [null],
+    [
+        "foo" => "bar",
+    ],
+]);
+var_dump($array);
 
 ?>
---EXPECT--
-O:13:"SplFixedArray":5:{i:0;s:3:"foo";i:1;N;i:2;i:42;i:3;O:8:"stdClass":1:{s:4:"prop";s:5:"value";}i:4;a:5:{i:0;i:1;i:1;i:2;i:2;i:3;i:3;i:4;i:4;i:5;}}
+--EXPECTF--
+Deprecated: Creation of dynamic property SplFixedArray::$foo is deprecated in %s on line %d
+O:13:"SplFixedArray":2:{i:0;a:5:{i:0;s:3:"foo";i:1;N;i:2;i:42;i:3;O:8:"stdClass":1:{s:4:"prop";s:5:"value";}i:4;a:5:{i:0;i:1;i:1;i:2;i:2;i:3;i:3;i:4;i:4;i:5;}}i:1;a:1:{s:3:"foo";s:3:"bar";}}
```

Comment:
As an example of my comment in ext/spl/spl_fixedarray.c - `__serialize` and `__unserialize` should be written in a way that ensures this line with the serialized output shouldn't change at all in the test case in this PR.

Otherwise, when users migrate from php <=8.1 to 8.2+, and are running both at the same time, serialized data structures containing SplFixedArrays(or subclasses of SplFixedArray)  will either unserialize differently in the other php version, or throw an Error